File: solutions/ruby/01-init/code/app/main.rb
def match_pattern(input_line, pattern)
  if pattern.length == 1
    input_line.include?(pattern)
  else
    raise "Unhandled pattern #{pattern}"
  end
end

def main
  pattern = ARGV[1]
  input_line = STDIN.gets

  if ARGV[0] != "-E"
    puts "Expected first argument to be '-E'"
    exit 1
  end

  if match_pattern(input_line, pattern)
    exit 0
  else
    exit 1
  end
end

main

File: solutions/typescript/01-init/code/app/main.ts
import { TextDecoder } from "util";

const args = process.argv;
const pattern = args[3];

const inputBuffer: ArrayBuffer = await Bun.stdin.arrayBuffer();
const decoder: TextDecoder = new TextDecoder();
const inputLine: string = decoder.decode(inputBuffer);

function matchPattern(inputLine: string, pattern: string): boolean {
  if (pattern.length === 1) {
    return inputLine.includes(pattern);
  } else {
    throw new Error(`Unhandled pattern: ${pattern}`);
  }
}

if (args[2] !== "-E") {
  console.log("Expected first argument to be '-E'");
  process.exit(1);
}

if (matchPattern(inputLine, pattern)) {
  process.exit(0);
} else {
  process.exit(1);
}
